### ISSUE TITLE
Require structure only for certain files / paths

### ISSUE THREAD

**Issue by pascalberger** [DavidAnson/markdownlint#32]
Is there a way to have [MD043](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md043---required-header-structure) only required for certain files or paths, but not for others?

Or is there a way to exclude this rule for certain files?

I tried with `<!-- markdownlint-disable MD043 -->` but couldn't get it working.

**Comment by DavidAnson** [comment]
Customizing the config object for `MD043` for each file is one approach. Another is turning `MD043` off via a comment as you show above. To verify this, I changed `example/good.md` to:

```md
# good.md

This file passes all rules.
```

And `example/standalone.js` to:

```js
"use strict";

var markdownlint = require("../lib/markdownlint");

var options = {
  "files": [ "good.md" ],
  "config": {
    "MD041": false,
    "MD043": {
      "headers": [
        "# header"
      ]
    }
  },
  "resultVersion": 1
};

var result = markdownlint.sync(options);
console.log(result.toString(true));
```

At this point, the output is:

```
good.md: 1: MD043/required-headers Required header structure [Expected: # header; Actual: # good.md]
```

Updating `good.md` to disable `MD043` with an inline comment looks like:

```md
<!-- markdownlint-disable MD043 -->

# good.md

This file passes all rules.
```

And, indeed, suppresses the `MD043` warning above.

Could you please double-check your configuration? Or tell me how to reproduce the problem?

Thanks!

**Comment by pascalberger** [comment]
Thanks, I got it working by suppressing `MD043` with a comment (but which needed at the end of the document, otherwise it would raise an [MD041](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md041---first-line-in-file-should-be-a-top-level-header) issue).
